haystack/nodes/prompt/providers.py

```diff
@@ -545,3 +547,111 @@ def supports(cls, model_name_or_path: str, **kwargs) -> bool:
         return (
             valid_model and kwargs.get("azure_base_url") is not None and kwargs.get("azure_deployment_name") is not None
         )
+
+
+class ChatGPTInvocationLayer(OpenAIInvocationLayer):
+    """
+    ChatGPT Invocation Layer
+
+    PromptModelInvocationLayer implementation for OpenAI's GPT-3 ChatGPT API. Invocations are made using REST API.
+    See [OpenAI ChatGPT API](https://platform.openai.com/docs/guides/chat) for more details.
+
+    Note: kwargs other than init parameter names are ignored to enable reflective construction of the class
+    as many variants of PromptModelInvocationLayer are possible and they may have different parameters.
+    """
+
+    def __init__(
+        self, api_key: str, model_name_or_path: str = "gpt-3.5-turbo", max_length: Optional[int] = 500, **kwargs
+    ):
+        super().__init__(api_key, model_name_or_path, max_length, **kwargs)
+
+    def invoke(self, *args, **kwargs):
+        """
+        It takes in either a prompt or a list of messages and returns a list of responses, using a REST invocation.
+
+        :return: A list of generated responses.
+
+        Note: Only kwargs relevant to OpenAI are passed to OpenAI rest API. Others kwargs are ignored.
+        For more details, see [OpenAI ChatGPT API reference](https://platform.openai.com/docs/api-reference/chat).
+        """
+        prompt = kwargs.get("prompt", None)
+
+        if isinstance(prompt, str):
+            messages = [{"role": "user", "content": prompt}]
+        elif isinstance(prompt, list) and len(prompt) > 0 and isinstance(prompt[0], dict):
+            messages = prompt
+        else:
+            raise ValueError(
+                f"The prompt format is different than what the model expects."
+                f"The model {self.model_name_or_path} requires either a string or messages in the ChatML format."
```

Comment:
Same here, add a space.
```suggestion
                f"The model {self.model_name_or_path} requires either a string or messages in the ChatML format. "
```